viewing_party/party.py

```diff
@@ -1,23 +1,182 @@
+        if movie["title"] not in friends_movies
+    ]
+
+    return unique_movies
+
+# Get movies unique to friends that user hasn't watched
+def get_friends_unique_watched(user_data):
+
+    # Put movie titles user has watched in a set
+    user_movie_titles = get_titles(user_data["watched"])
+
+    # Create a list for movies only friends watched
+    unique_friends_movies = []
+
+    # Put movies friends watched to a set
+    friends_movie_titles = set()
+   
+    # Iterate through movies, if movie title is not in a set 
+    # unique to a user and if movie title is not in friends movies
+    # add it to a set of titles unique to friends, so that we don't add
+    # duplicates to unique friend's movies. Append movies 
+    # to a list of unique movies 
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            title = movie["title"]
+            if title not in user_movie_titles:
+                if title not in friends_movie_titles:
+                    friends_movie_titles.add(title)
+                    unique_friends_movies.append(movie)
+
+    return unique_friends_movies
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+# Get friends' recommendations for user
+def get_available_recs(user_data):
+
+    # Get unique movies friends watched, but user hasn't
+    unique_movies = get_friends_unique_watched(user_data)
+
+    recommended_movies = []
+    # Append a movie to a list of recommended movies
+    # if a movie is hosted by a subscription user has acces to
+    for movie in unique_movies:
+        host = movie["host"]
+        if host in user_data["subscriptions"]:
+            recommended_movies.append(movie)
+    return recommended_movies
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+# Recommend movies from friends list based on user most frequent genre
+def get_new_rec_by_genre(user_data):
+
+    # Get user most frequent genre
+    most_frequent_genre = get_most_watched_genre(user_data)
+
+    # Create list of recommended movies with most frequent genre
+    not_watched = get_friends_unique_watched(user_data)
+    recommended_movies = [
+        movie for movie in not_watched 
+        if movie["genre"] == most_frequent_genre
+    ]
+
+    return recommended_movies
+
+# Recommend movies from user favorites
+def get_rec_from_favorites(user_data):
+    # Get the list of movies that only the user has watched
+    unique_movies = get_unique_watched(user_data)
+
+    # Get a set of favorite movie titles
+    favorite_movies = get_titles(user_data["favorites"])
+    
+    # Filter the unique movies to only those that are in the user's favorites
+    recommended_favorites = [
+        movie
+        for movie in unique_movies
+        if movie["title"] in favorite_movies
+    ]
+
+    return recommended_favorites

```

Comment:
This function looks really great too! Good job using methods from before and as always, another good list comprehension!